lib/src/view/study/study_list_screen.dart

```diff
@@ -0,0 +1,472 @@
+              const SizedBox(height: 10.0),
+            ],
+            Filter<StudyListOrder>(
+              // TODO l10n
+              filterName: 'Sort by',
+              filterType: FilterType.singleChoice,
+              choices: StudyListOrder.values,
+              choiceSelected: (choice) => filter.order == choice,
+              choiceLabel: (order) => Text(studyListOrderL10n(order, context)),
+              onSelected: (value, selected) =>
+                  ref.read(studyFilterProvider.notifier).setOrder(value),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _Body extends StatefulWidget {
+  const _Body({
+    required this.filter,
+  });
+
+  final StudyFilterState filter;
+
+  @override
+  State<_Body> createState() => _BodyState();
+}
+
+class _BodyState extends State<_Body> {
+  String? search;
+
+  final _searchController = SearchController();
+
+  @override
+  void dispose() {
+    super.dispose();
+    _searchController.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SafeArea(
+      child: Column(
+        children: [
+          Padding(
+            padding: Styles.bodySectionPadding,
+            child: SearchBar(
+              controller: _searchController,
+              leading: const Icon(Icons.search),
+              trailing: [
+                if (search != null)
+                  IconButton(
+                    onPressed: () => setState(() {
+                      search = null;
+                      _searchController.clear();
+                    }),
+                    tooltip: 'Clear',
+                    icon: const Icon(
+                      Icons.close,
+                    ),
+                  ),
+              ],
+              hintText: search ?? context.l10n.searchSearch,
+              onSubmitted: (term) {
+                setState(() {
+                  search = term;
+                });
+              },
+            ),
+          ),
+          _StudyList(
+            paginatorProvider: StudyListPaginatorProvider(
+              filter: widget.filter,
+              search: search,
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _StudyList extends ConsumerStatefulWidget {
+  const _StudyList({
+    required this.paginatorProvider,
+  });
+
+  final StudyListPaginatorProvider paginatorProvider;
+
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() => _StudyListState();
+}
+
+class _StudyListState extends ConsumerState<_StudyList> {
+  final _scrollController = ScrollController();
+
+  @override
+  void initState() {
+    super.initState();
+    _scrollController.addListener(_scrollListener);
+  }
+
+  @override
+  void dispose() {
+    _scrollController.removeListener(_scrollListener);
+    _scrollController.dispose();
+    super.dispose();
+  }
+
+  void _scrollListener() {
+    if (_scrollController.position.pixels ==
```

Comment:
I think we should trigger the next page before the scroll reaches the end, so the scrolling will feel smoother. I know it's not done like that in other paginated view but that's something I wanted to change (will write an issue).

I don't know what is a good threshold, perhaps 75% of the max scroll extent?

Reply:
75% sounds good :)
 
I had to introduce an additional `bool` to the state to avoid triggering the request multiple times before the next page is loaded.